src/main/java/core/basesyntax/service/ShopServiceImpl.java

```
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
```

Constructor accepts `OperationStrategy` but doesn't validate it. Add a null check and throw an informative RuntimeException if `operationStrategy` is null to fail fast and respect dependency invariants.